frontend/views/components/InviteLinkToCopy.vue

```diff
@@ -0,0 +1,134 @@
+<template lang='pug'>
+component.c-wrapper(
+  :is='tag'
+  @click='copyToClipboard'
+  data-test='invitationLink'
+)
+  input.c-invisible-input(
+    type='text'
+    ref='input'
+    :value='inviteLink'
+  )
+  span.link.has-ellipsis.c-link {{ inviteLink }}
+  button.is-icon-small.has-background.c-copy-button(:aria-label='L("Copy link")')
+    i.icon-copy.is-regular
+  transition(name='fade')
+    i18n.c-tooltip(
+      v-if='ephemeral.isTooltipActive'
+      :style='ephemeral.tooltipPosition'
+      key='test'
+      tag='p'
+      v-append-to-body=''
+    ) Copied to clipboard!
+</template>
+
+<script>
+export default {
+  name: 'InviteLinkToCopy',
+  data: () => ({
+    ephemeral: {
+      isTooltipActive: false,
+      tooltip: null,
+      tooltipPosition: null
+    }
+  }),
+  props: {
+    inviteLink: {
+      type: String,
+      required: true
+    },
+    tag: {
+      type: String,
+      default: 'div'
+    }
+  },
+  methods: {
+    copyToClipboard () {
+      // if the user is using the device that supports web share API, use it and then skip the other logics below.
+      if (navigator.share) {
+        navigator.share({
+          title: this.L('Your invite'),
+          url: this.textToCopy
+        })
+        return
+      }
+
+      this.$refs.input.select()
+      document.execCommand('copy')
+      this.ephemeral.isTooltipActive = true
+      setTimeout(() => {
+        this.ephemeral.isTooltipActive = false
+      }, 1500)
+    },
+    adjustPosition () {
+      this.trigger = this.$el.getBoundingClientRect()
+      const { scrollX, scrollY } = window
+      const { width, left, top } = this.trigger
+      const spacing = 16
+      const x = scrollX + left + width / 2 - this.ephemeral.tooltip.width / 2
+      const y = scrollY + top - (this.ephemeral.tooltip.height + spacing)
+
+      this.ephemeral.tooltipPosition = `transform: translate(${x}px, ${y}px)`
+    }
+  },
+  directives: {
+    appendToBody: {
+      // refer to the comment in Tooltip.vue for the further information on this custom directive.
```

Comment:
If this is identical code to what's in Tooltip.vue, then to DRY it, can you put `appendToBody` in a Vue mixin and import it here and in Tooltip.vue?

And I am wondering also if the same applies to `adjustPosition`?

Reply:
The issue here(along with another issue [here](https://github.com/okTurtles/group-income-simple/pull/765#discussion_r344121664)) is fixed by using the existing `Tooltip.vue` component. That way we can DRY it without creating extra Vue mixin file. :)